addons/mail/models/mail_thread.py

```diff
@@ -2582,9 +2582,12 @@ def _notify_classify_recipients(self, recipient_data, model_name):
         only return groups with recipients
         """
 
+        if msg_vals is False:
+            msg_vals = {}
```

Comment:
Is that function called in a lot of places with `msg_vals=False` ? If yes, then you should use it as default value instead of None, if no then the function should never be called with `msg_vals=False` and the wrong call(s) should be modified to use `None` or don't provide any value

Reply:
Nope because setting a dict as default param value makes it shared. Mutables objects may sometimes lead to funny errors :) . When giving a dict as optional value, better set it as None as default and instantiate a void one if really necessary afterwards.

Reply:
That's what I wanted to say : use `None` as default param and never call with False or use False as default param (sorry if it wasn't clear)

Reply:
True !